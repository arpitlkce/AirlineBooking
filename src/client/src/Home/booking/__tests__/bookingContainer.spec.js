import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import BookingContainer from '../bookingContainer';
const mockStore = configureMockStore();
const store = mockStore({
    itemSearchFetchSucceeded: [{
        'flightData': {
            'flightName': 'Go Air',
            'flightNumber': 'G8-428'
        },
        'departData': {
            'departureTime': '05:35',
            'departureAirport': 'New Delhi'
        },
        'arrivalData': {
            'arrivalTime': '08:35',
            'arrivalAirport': 'Bangalore'
        },

        'duration': '2:30',
        'price': '10,320',
        'departureAirportCode': 'DEL',
        'arrivalAirportCode': 'BLR'
    }]
});

describe('Booking Container Component', () => {
    it('should render correctly in  mode', () => {
        const component = renderer.create(<Provider store={store}><BookingContainer /></Provider>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
