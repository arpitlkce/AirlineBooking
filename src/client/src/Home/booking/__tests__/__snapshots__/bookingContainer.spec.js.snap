// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking Container Component should render correctly in  mode 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      Go Fly
    </a>
    <ul
      className="mar-auto navbar-nav"
    >
      <li
        className="nav-item"
      >
        <h3>
           Welcome to Go Fly 
        </h3>
      </li>
    </ul>
    <button
      aria-label={null}
      className="btn btn-light"
      onClick={[Function]}
    >
      Log Out
    </button>
  </nav>
  <div
    className="container"
  >
    <div
      className="jumbotron jumbotron-fluid"
    >
      <div
        className="container-fluid"
      >
        <div
          className="lead row"
        >
          <div
            className="col"
          >
            <label
              className=""
            >
              Flight Details 
            </label>
          </div>
          <div
            className="col"
          >
            <label
              className=""
            >
              Departure Details
            </label>
          </div>
          <div
            className="col"
          >
            <label
              className=""
            >
              Arrival Details
            </label>
          </div>
          <div
            className="col"
          >
            <label
              className=""
            >
              Duration
            </label>
          </div>
          <div
            className="col"
          >
            <label
              className=""
            >
              Price
            </label>
          </div>
        </div>
        <div
          className="small lead row"
        >
          <div
            className="col"
          >
            <div>
              Go Air
            </div>
            <div>
              G8-428
            </div>
          </div>
          <div
            className="col"
          >
            <div>
              05:35
            </div>
            <div>
              DEL
            </div>
            <div>
              New Delhi
            </div>
          </div>
          <div
            className="col"
          >
            <div>
              08:35
            </div>
            <div>
              BLR
            </div>
            <div>
              Bangalore
            </div>
          </div>
          <div
            className="col"
          >
            <div>
              2:30
               Hrs
            </div>
          </div>
          <div
            className="col"
          >
            <div>
              $ 
              10,320
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="form-row"
    >
      <div
        className="col-md-4"
      />
      <div
        className="col-md-4"
      />
      <div
        className="col-md-4"
      />
    </div>
    <div
      className="form-row"
    >
      <div
        className="col-md-6"
      >
        <button
          aria-label={null}
          className="btn btn-secondary"
          onClick={[Function]}
          type="button"
        >
          Book
        </button>
      </div>
    </div>
  </div>
</div>
`;
